src/Security/RootPermissions.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace SilverStripe\Security;
+
+/**
+ * For root objects that have permission type "Inherit" determine the result of certain permission checks
```

Comment:
I think RootPermissions is incorrect, as the permissions actually accounts for the -1 level. "root" permissions applies to all items at the root, which are actual instances of DataObjects.

Consider this handler the "imaginary number" equivalent of permissions, which attempts to generalise permissions for objects outside the natural bounds within which objects can be created (root, child, nested). This is why it doesn't apply directly to permissions for actual DataObject instances.